#
# #define CLOCK_BOOTTIME since it's not present in our glibc
#
--- a/src/unix/linux.c
+++ b/src/unix/linux.c
@@ -56,6 +56,10 @@
 #include <time.h>
 #include <unistd.h>
 
+#ifndef CLOCK_BOOTTIME
+# define CLOCK_BOOTTIME 7
+#endif
+
 #ifndef __NR_io_uring_setup
 # define __NR_io_uring_setup 425
 #endif
#
# Lightweight patch to omit sendmmsg/recvmmsg, as it's not present in our glibc
# and it's only used for udp
#
--- a/src/unix/udp.c
+++ b/src/unix/udp.c
@@ -152,6 +152,7 @@ static void uv__udp_io(uv_loop_t* loop, uv__io_t* w, unsigned int revents) {
 }
 
 static int uv__udp_recvmmsg(uv_udp_t* handle, uv_buf_t* buf) {
+#if !defined(NO_MMSG)
 #if defined(__linux__) || defined(__FreeBSD__) || defined(__APPLE__)
   struct sockaddr_in6 peers[20];
   struct iovec iov[ARRAY_SIZE(peers)];
@@ -218,6 +219,9 @@ static int uv__udp_recvmmsg(uv_udp_t* handle, uv_buf_t* buf) {
 #else  /* __linux__ || ____FreeBSD__ || __APPLE__ */
   return UV_ENOSYS;
 #endif  /* __linux__ || ____FreeBSD__ || __APPLE__ */
+#else
+  return UV_ENOSYS;
+#endif
 }
 
 static void uv__udp_recvmsg(uv_udp_t* handle) {
@@ -890,9 +894,11 @@ int uv__udp_init_ex(uv_loop_t* loop,
 
 
 int uv_udp_using_recvmmsg(const uv_udp_t* handle) {
+#if !defined(NO_MMSG)
 #if defined(__linux__) || defined(__FreeBSD__) || defined(__APPLE__)
   if (handle->flags & UV_HANDLE_UDP_RECVMMSG)
     return 1;
+#endif
 #endif
   return 0;
 }
@@ -1311,6 +1317,7 @@ static int uv__udp_sendmsgv(int fd,
   r = 0;
   nsent = 0;
 
+#if !defined(NO_MMSG)
 #if defined(__linux__) || defined(__FreeBSD__) || defined(__APPLE__) || \
   (defined(__sun__) && defined(MSG_WAITFORONE))
   if (count > 1) {
@@ -1342,6 +1349,7 @@ static int uv__udp_sendmsgv(int fd,
 #endif  /* defined(__linux__) || defined(__FreeBSD__) || defined(__APPLE__) ||
 	 * (defined(__sun__) && defined(MSG_WAITFORONE))
 	 */
+#endif
 
   for (i = 0; i < count; i++, nsent++)
     if ((r = uv__udp_sendmsg1(fd, bufs[i], nbufs[i], addrs[i])))
